include/MotorPlugin.hh

```diff
@@ -0,0 +1,67 @@
+/*
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef MOTORPLUGIN_HH_
+#define MOTORPLUGIN_HH_
+
+#include <memory>
+
+#include <gz/sim/System.hh>
+
+namespace gz {
+namespace sim {
+inline namespace GZ_SIM_VERSION_NAMESPACE {
+namespace systems {
+
+
+class MotorPlugin :
+    public System,
+    public ISystemPreUpdate,
+    public ISystemConfigure
+{
+	/// \brief Destructor
+	public: virtual ~MotorPlugin();
+
+	/// \brief Constructor
+	public: MotorPlugin();
+
+	// Documentation inherited
+	public: void PreUpdate(const gz::sim::UpdateInfo &_info,
+							EntityComponentManager &_ecm) final;
+
+	// Documentation inherited
+	public: void Configure(const Entity &_entity,
+							const std::shared_ptr<const sdf::Element> &_sdf,
+							EntityComponentManager &_ecm,
+							EventManager &) final;
+
+	/// \brief Load control channels
+	private: void LoadControlChannels(
+		sdf::ElementPtr _sdf,
+		gz::sim::EntityComponentManager &_ecm);
```

Comment:
Can be in the private `Impl` class as this function is not called by the Gazebo system.